### PR TITLE
feat: doctor command refactor

### PR DESCRIPTION
**Pull request by radekdymacz**
Fixes #

## Checklist

- [x] Documented

### CODE REVIEW

File: metrics/deploy/lib/doctor/doctor_command.dart [hunk 1]
```diff
@@ -10,12 +14,18 @@ class DoctorCommand extends Command {
   @override
   final description = "Check dependencies.";
 
+  final _firebase = FirebaseCommand();
+  final _gcloud = GCloudCommand();
+  final _git = GitCommand();
+  final _fluter = FlutterCommand();
```
Comment (bootstraponline) [inline metrics/deploy/lib/doctor/doctor_command.dart:20]: I think this is a typo?

Comment (radekdymacz) [reply metrics/deploy/lib/doctor/doctor_command.dart:20]: oops will fix that

File: metrics/deploy/lib/cli/firebase/firebase_command.dart [hunk 2]
```diff
@@ -73,4 +73,9 @@ class FirebaseCommand {
         ],
         verbose: true);
   }
+
+  /// Print cli verison.
```
Comment (solid-vovabeloded) [inline metrics/deploy/lib/cli/firebase/firebase_command.dart:77]: Please, start the method documentation with third-person verbs as [effective dart](https://dart.dev/guides/language/effective-dart/documentation#prefer-starting-function-or-method-comments-with-third-person-verbs) advises. The same suggestion for methods in the `FlutterCommand`, `GcloudCommand`, `GitCommand`, and `DoctorCommand`. 
```suggestion
  /// Prints CLI version.
```

File: metrics/deploy/lib/cli/flutter/flutter_command.dart [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import 'package:process_run/process_run.dart' as cmd;
+
+/// class wrapping up flutter CLI
```
Comment (solid-vovabeloded) [inline metrics/deploy/lib/cli/flutter/flutter_command.dart:3]: Please, format the documentation as sentences (see [effective dart](https://dart.dev/guides/language/effective-dart/documentation#do-format-comments-like-sentences) suggestion). The same for all the rest classes.
```suggestion
/// A wrapper class for the Flutter CLI.
```

File: metrics/deploy/lib/cli/git/git_command.dart [hunk 4]
```diff
@@ -2,7 +2,13 @@ import 'package:process_run/process_run.dart' as cmd;
 
 /// class wrapping up gcloud CLI
 class GitCommand {
+  /// Git clone command.
```
Comment (solid-vovabeloded) [inline metrics/deploy/lib/cli/git/git_command.dart:5]: Would it be more clear?
```suggestion
  /// Clones the git repository from the given [repoURL] to the given [srcPath].
```

**Review by solid-vovabeloded — CHANGES_REQUESTED** [review]
The changes look great, @radekdymacz! I've added a couple of suggestions considering documentation - please take a look.

File: metrics/deploy/lib/cli/firebase/firebase_command.dart [hunk 5]
```diff
@@ -1,18 +1,19 @@
 import 'package:process_run/process_run.dart' as cmd;
 import 'package:process_run/shell.dart';
 
-/// class wrapping up firebase CLI
+/// A wrapper class for the Firebase CLI.
 class FirebaseCommand {
-  /// Login to GCloud and Firebase and get firebase CI token
+  /// Logins to GCloud and Firebase and gets firebase CI token.
```
Comment (solid-vovabeloded) [inline metrics/deploy/lib/cli/firebase/firebase_command.dart:6]:
```suggestion
  /// Logins to GCloud and Firebase and gets the Firebase CI token.
```

File: metrics/deploy/lib/cli/gcloud/gcloud_command.dart [hunk 6]
```diff
@@ -3,24 +3,29 @@ import 'dart:math';
...
+  /// The unique identifier of the project.
   String projectID = '';
 
-  /// Generates random string for new project name
+  /// Generates a random string for the new project name.
   String _getRandomString(int length) => String.fromCharCodes(Iterable.generate(
       length, (_) => _chars.codeUnitAt(_rnd.nextInt(_chars.length))));
 
-  /// Login to GCloud and Firebase and get firebase CI token
+  /// Logins to GCloud and Firebase and gets firebase CI token.
```
Comment (solid-vovabeloded) [inline metrics/deploy/lib/cli/gcloud/gcloud_command.dart:21]:
```suggestion
  /// Logins to GCloud and Firebase and gets a firebase CI token.
```

File: metrics/deploy/lib/cli/gcloud/gcloud_command.dart [hunk 7]
```diff
@@ -37,7 +42,7 @@ class GCloudCommand {
     return projectID;
   }
 
-  // Add project app needed to create firestore database.
+  /// Adds project app needed to create firestore database.
```
Comment (solid-vovabeloded) [inline metrics/deploy/lib/cli/gcloud/gcloud_command.dart:45]:
```suggestion
  /// Adds project app needed to create a Firestore database.
```

File: metrics/deploy/lib/cli/gcloud/gcloud_command.dart [hunk 8]
```diff
@@ -3,24 +3,29 @@ import 'dart:math';
...
+  /// Logins to GCloud and Firebase and gets firebase CI token.
   Future<void> login() async {
-    // GCloud login
+    // Logins to GCloud.
     print('GCloud Login.');
     await cmd.run('gcloud', ['auth', 'login'], verbose: true);
   }
 
-  /// Add project or use existing project
+  /// Adds project or uses existing project.
```
Comment (solid-vovabeloded) [inline metrics/deploy/lib/cli/gcloud/gcloud_command.dart:28]:
```suggestion
  /// Adds project or uses an existing one.
```

File: metrics/deploy/lib/cli/gcloud/gcloud_command.dart [hunk 9]
```diff
@@ -49,7 +54,7 @@ class GCloudCommand {
     }
   }
 
-  /// Create firestore database.
+  /// Creates firestore database with the given [region] and [projectID].
```
Comment (solid-vovabeloded) [inline metrics/deploy/lib/cli/gcloud/gcloud_command.dart:57]:
```suggestion
  /// Creates a Firestore database with the given [region] and [projectID].
```

File: metrics/deploy/lib/doctor/doctor_command.dart [hunk 10]
```diff
@@ -1,21 +1,36 @@
...
 
-/// class providing doctor command to verify dependencies.
+/// A class providing doctor command to verify dependencies.
 class DoctorCommand extends Command {
   @override
   final name = "doctor";
   @override
   final description = "Check dependencies.";
 
+  /// Provides Firebase CLI.
```
Comment (solid-vovabeloded) [inline metrics/deploy/lib/doctor/doctor_command.dart:14]:
```suggestion
  /// A [FirebaseCommand] needed to get the Firebase CLI version.
```
The same for all fields below.

File: metrics/deploy/lib/doctor/doctor_command.dart [hunk 11]
```diff
@@ -1,21 +1,36 @@
...
+  /// Provides GCloud CLI.
+  final _gcloud = GCloudCommand();
+
+  /// Provides Git CLI.
+  final _git = GitCommand();
+
+  /// Provides Flutter CLI.
+  final _flutter = FlutterCommand();
+
+  /// Creates this doctor command instance.
```
Comment (solid-vovabeloded) [inline metrics/deploy/lib/doctor/doctor_command.dart:26]: Maybe we can remove this constructor since the class has an empty constructor by default?

**Review by solid-vovabeloded — CHANGES_REQUESTED** [review]
Great job so far, @radekdymacz & @solid-tetianamovlian! I've added a couple of suggestions here - please take a look.

**Review by solid-vovabeloded — APPROVED** [review]
LGTM, thanks @radekdymacz & @solid-tetianamovlian! 🚀

**Pull request merged**